packages/app-page-builder-elements/package.json

```diff
@@ -17,6 +17,7 @@
     "@babel/runtime": "^7.22.6",
     "@emotion/react": "^11.10.6",
     "@emotion/styled": "^11.10.6",
+    "@webiny/app-dynamic-pages": "0.0.0",
```

Comment:
My main concern with this line here is that this package introduces a whole bucket of new dependencies, which are potentially gonna get bundled and bloat the public website code. Which of course then affects the SEO and all that related stuff.

I checked what's the usage of the package, and noticed we're using it because we need `useDynamicData` hook. 

So, what I'm thinking are two things.

We extract the React hook and related code into a separate package, like `app-dynamic-pages-hooks`. That package would not include any UI nor all of the big and irrelevant deps in its package.json, like `@emotion/styled`, `@fortawesome/react-fontawesome`, or the huge `@webiny/ui`.

The `app-page-builder-elements` package would then include that, and would no longer include `app-dynamic-pages`.

We could take this even further and not have the new package in `app-page-builder-elements` at all. We can simply adapt the same approach that's used  for example here: `packages/app-page-builder/src/render/plugins/elements/button/index.tsx`.

Basically, instead of hardcoding deps into page elements, we allow them do be passed on the element construction. For example, with the `linkComponent` prop, we allow users to pass a custom Link component. This will never be needed if users are using Webiny and its website rendering capabilities, but if a user is using, say, Next.js to render a page, then here they would most probably want to pass Next.js's Link component. This approach will also prevent things like Apollo React client being installed in external projects, and allow users to implement their own way of fetching data, if needed.

I hope I'm making sense. I know some of these are NTH, and I'd say that at this moment, let's try to maybe implement at least the first part. That way we're making at least some effort in preventing the website bundle to get even larger that it already is. At the moment, there's really a lot of extra stuff that degrades the overall performance and SEO score of the public webste, we'll need to address this sometime in the future. I can see it when dynamic pages gets out, and people start using more PB to render their website.

We can consider the 2nd part as a NTH.

Let me know if a call is needed potentially.